openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/BotRepository.java

```diff
@@ -41,6 +41,10 @@ public Bot setFields(Bot entity, Fields fields) throws IOException {
   @Override
   public void prepare(Bot entity) throws IOException {
     User user = daoCollection.userDAO().findEntityById(entity.getBotUser().getId(), Include.ALL);
+    if (!Boolean.TRUE.equals(user.getIsBot())) {
+      throw new IllegalArgumentException(
+          String.format("User [%s] can be assigned to a Bot since isBot flag is not set", user.getName()));
```

Comment:
I believe the error message should be `User [%s] cannot be assigned [...]` vs `User [%s] can be assigned [...]`